pkgs/applications/audio/csound/default.nix

```
@@ -26,21 +30,32 @@ stdenv.mkDerivation rec {
   };
 
   cmakeFlags = [ "-DBUILD_CSOUND_AC=0" ] # fails to find Score.hpp
-    ++ lib.optional (libjack2 != null) "-DJACK_HEADER=${libjack2}/include/jack/jack.h";
+               ++ lib.optional stdenv.isDarwin "-DCS_FRAMEWORK_DEST=${placeholder "out"}/lib"
+               ++ lib.optional (libjack2 != null) "-DJACK_HEADER=${libjack2}/include/jack/jack.h";
```

```suggestion
    ++ lib.optional stdenv.isDarwin "-DCS_FRAMEWORK_DEST=${placeholder "out"}/lib"
    ++ lib.optional (libjack2 != null) "-DJACK_HEADER=${libjack2}/include/jack/jack.h";
```
Please don't rely on overwriting packages with null which is an anti pattern.

---

100% valid suggestion there. But I'm not entirely sure what you mean overwriting packages with null. At least there's no null assignment?

---

What is done here https://github.com/NixOS/nixpkgs/pull/171993/files#diff-4ed2a92bbf6e18ddb3330dae0f0c89e4a70c3392d5e9ae24578d4cf3475c8a9dR6-R16 is a big anti pattern

---

ok happy to fix it (it was this way before). I assume they need to be applied using booleans?

---

yeah but that is a bit out of scope for this PR.